### PR TITLE
[UI] Initial implement the SQL Lab page

### PR DESCRIPTION
**Pull request by labbomb**
### _Why are the changes needed?_

1. New SQL Lab Page
2. New core editor functionality
image

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No.

### CODE REVIEW

**Comment by yaooqinn** [comment]
Wow, that's really impressive!

**Comment by zhaohehuhu** [comment]
good job. Thanks.

**Comment by cfmcgrady** [comment]
👍

File: kyuubi-server/web-ui/package.json [hunk 1]
```diff
@@ -19,8 +19,10 @@
     "axios": "^0.27.2",
     "date-fns": "^2.29.3",
     "element-plus": "^2.2.12",
+    "monaco-editor": "^0.44.0",
```
Comment (pan3793) [inline kyuubi-server/web-ui/package.json:22]: the ASF policy requires listing dips at LICENSE and NOTICE, for this case, please run `pnpm licenses list --prod` and update the content of `LICENSE-binary`

File: kyuubi-server/web-ui/pnpm-lock.yaml [hunk 2]
```diff
@@ -1,9 +1,5 @@
 lockfileVersion: '6.0'
 
-settings:
-  autoInstallPeers: true
-  excludeLinksFromLockfile: false
```
Comment (pan3793) [inline kyuubi-server/web-ui/pnpm-lock.yaml:5]: how can we benefit from this change?

File: kyuubi-server/web-ui/src/layout/components/aside/types.ts [hunk 3]
```diff
@@ -94,5 +94,10 @@ export const MENUS = [
     label: 'Contact Us',
     icon: 'PhoneFilled',
     router: '/contact'
+  },
+  {
+    label: 'SQL Lab',
+    icon: 'Cpu',
```
Comment (pan3793) [inline kyuubi-server/web-ui/src/layout/components/aside/types.ts:100]: kind of an awkward expression... does "CPU"(prefered) or "cpu" work?

File: kyuubi-server/web-ui/src/components/monaco-editor/index.vue [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  const init = () => {
+    monaco.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
+      noSemanticValidation: true,
+      noSyntaxValidation: false
+    })
+    monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
+      target: monaco.languages.typescript.ScriptTarget.ES2020,
+      allowNonTsExtensions: true
+    })
+    monaco.languages.registerCompletionItemProvider('sql', {
```
Comment (pan3793) [inline kyuubi-server/web-ui/src/components/monaco-editor/index.vue:77]: must we write such a completion hepler by hand?

File: kyuubi-server/web-ui/src/components/monaco-editor/type.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    type: [String, Number] as PropType<string | number>,
+    default: '100%'
+  },
+  height: {
+    type: [String, Number] as PropType<string | number>,
+    default: '100%'
+  },
+  language: {
+    type: String as PropType<string>,
+    default: 'javascript'
```
Comment (pan3793) [inline kyuubi-server/web-ui/src/components/monaco-editor/type.ts:53]: emm,,. what does this represent? the default language of content of SQL editor?

File: kyuubi-server/web-ui/src/layout/components/aside/types.ts (hunk 3 above)
Comment (labbomb) [reply kyuubi-server/web-ui/src/layout/components/aside/types.ts:100]: This is the icon name provided by elp

File: kyuubi-server/web-ui/src/components/monaco-editor/index.vue (hunk 4 above)
Comment (labbomb) [reply kyuubi-server/web-ui/src/components/monaco-editor/index.vue:77]: Never mind, this is just a simple version of sql functionality

File: kyuubi-server/web-ui/src/layout/components/aside/types.ts (hunk 3 above)
Comment (pan3793) [reply kyuubi-server/web-ui/src/layout/components/aside/types.ts:100]: Alright, let's keep it as-is

File: kyuubi-server/web-ui/src/components/monaco-editor/index.vue (hunk 4 above)
Comment (pan3793) [reply kyuubi-server/web-ui/src/components/monaco-editor/index.vue:77]: I thought there may be a popular third-party implementation, seems not ...

File: kyuubi-server/web-ui/package.json (hunk 1 above)
Comment (labbomb) [reply kyuubi-server/web-ui/package.json:22]: ok

File: kyuubi-server/web-ui/pnpm-lock.yaml (hunk 2 above)
Comment (pan3793) [reply kyuubi-server/web-ui/pnpm-lock.yaml:5]: np, i found the new generated lock file just moves it to the end

File: kyuubi-server/web-ui/src/components/monaco-editor/index.vue [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  const codeEditBox = ref()
+  const init = () => {
+    monaco.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
+      noSemanticValidation: true,
+      noSyntaxValidation: false
+    })
+    monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
+      target: monaco.languages.typescript.ScriptTarget.ES2020,
+      allowNonTsExtensions: true
+    })
```
Comment (pan3793) [inline kyuubi-server/web-ui/src/components/monaco-editor/index.vue:64]: are they required?

File: kyuubi-server/web-ui/src/components/monaco-editor/index.vue [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  import jsonWorker from 'monaco-editor/esm/vs/language/json/json.worker?worker'
+  import EditorWorker from 'monaco-editor/esm/vs/editor/editor.worker?worker'
+  import { editorProps } from './type'
+  import { useEditorStore } from '@/pinia/editor'
+  import { ref, toRaw, watch, onBeforeUnmount, onMounted } from 'vue'
+
+  // @ts-ignore: worker
+  self.MonacoEnvironment = {
+    getWorker(_: string, label: string) {
+      if (label === 'json') {
```
Comment (pan3793) [inline kyuubi-server/web-ui/src/components/monaco-editor/index.vue:35]: necessary?

File: kyuubi-server/web-ui/src/components/monaco-editor/index.vue [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        const sqlStr = [
+          'select',
+          'from',
+          'where',
+          'and',
+          'or',
+          'limit',
+          'order by',
+          'group by'
+        ]
```
Comment (pan3793) [inline kyuubi-server/web-ui/src/components/monaco-editor/index.vue:84]: Are the keywords case-sensitive? if not
```suggestion
        const keywords = [
          'SELECT',
          'FROM',
          'WHERE',
          'AND',
          'OR',
          'LIMIT',
          'ORDER BY',
          'GROUP BY'
        ]
```

File: kyuubi-server/web-ui/package.json (hunk 1 above)
Comment (pan3793) [reply kyuubi-server/web-ui/package.json:22]: Thanks for updating, the newly introduced deps are all under Catalog A, which looks great.

https://www.apache.org/legal/resolved.html#asf-3rd-party-license-policy

**Review by pan3793 — APPROVED** [review]
LGTM, thanks for your excellent work!

**Comment by pan3793** [comment]
Merged to master for 1.9

**Pull request closed without merging**